h2o-genmodel/src/main/java/hex/genmodel/ModelMojoReader.java

```diff
@@ -47,14 +49,31 @@ public static MojoModel readFrom(MojoReaderBackend reader) throws IOException {
   /**
    * Retrieve value from the model's kv store which was previously put there using `writekv(key, value)`. We will
    * attempt to cast to your expected type, but this is obviously unsafe. Note that the value is deserialized from
-   * the underlying string representation using {@link ParseUtils#tryParse(String)}, which occasionally may get the
-   * answer wrong.
+   * the underlying string representation using {@link ParseUtils#tryParse(String, Object)}, which occasionally may get
+   * the answer wrong.
    * If the `key` is missing in the local kv store, null will be returned. However when assigning to a primitive type
    * this would result in an NPE, so beware.
    */
   @SuppressWarnings("unchecked")
   protected <T> T readkv(String key) {
-    return (T) _lkv.get(key);
+    return (T) readkv(key, null);
+  }
+
+  /**
+   * Retrieves the value associated with a given key. If value is not set of the key, a given default value is returned
+   * instead. Uses same parsing logic as {@link ModelMojoReader#readkv(String)}. If default value is not null it's type
+   * is used to assist the parser to determine the return type.
+   * @param key name of the key
+   * @param defVal default value
+   * @param <T> return type
+   * @return parsed value
+   */
+  @SuppressWarnings("unchecked")
+  protected <T> T readkv(String key, T defVal) {
```

Comment:
The `readkv` with a default can be implemented simply as
    
    Object value = _lkv.get(key);
    return value == null? defaultValue : (T) value;

and then you don't need all this complexity with `RawValue`s.

However we have not had a need for such a function so far (conceivably it can be useful for reading old mojos after the format of the mojo has substantially evolved).